tests/models/graph_tests.py

```diff
@@ -1481,3 +1271,1001 @@ def test_add_resource_instance_lifecycle(self):
+            graphid=self.NODE_NODETYPE_GRAPHID,
+        )
+        editable_future_graph.save()
+
+        updated_source_graph = source_graph.update_from_editable_future_graph(
+            editable_future_graph=editable_future_graph
+        )
+        editable_future_graph = updated_source_graph.create_editable_future_graph()
+
+        node = [node for node in editable_future_graph.nodes.values()][2]
+
+        # fixes flaky test
+        models.NodeGroup.objects.filter(pk=node.pk).delete()
+
+        nodegroup_count_before = models.NodeGroup.objects.count()
+
+        source_identifier_id = node.source_identifier_id
+        original_nodegroup_id = node.nodegroup_id
+        updated_nodegroup_id = node.pk
+
+        models.NodeGroup.objects.create(
+            **{
+                "cardinality": "n",
+                "legacygroupid": "",
+                "nodegroupid": str(updated_nodegroup_id),
+                "parentnodegroup_id": None,
+            }
+        ).save()
+
+        node.nodegroup_id = updated_nodegroup_id
+        node.save()
+
+        updated_source_graph = updated_source_graph.update_from_editable_future_graph(
+            editable_future_graph=editable_future_graph
+        )
+        editable_future_graph = updated_source_graph.create_editable_future_graph()
+
+        # a source_graph nodegroup and an editable_future_graph nodegroup have been created
+        self.assertEqual(nodegroup_count_before, models.NodeGroup.objects.count() - 2)
+
+        node = [
+            node
+            for node in editable_future_graph.nodes.values()
+            if node.source_identifier_id == source_identifier_id
+        ][0]
+        node.nodegroup_id = original_nodegroup_id
+        node.save()
+
+        updated_source_graph = updated_source_graph.update_from_editable_future_graph(
+            editable_future_graph=editable_future_graph
+        )
+        editable_future_graph = updated_source_graph.create_editable_future_graph()
+
+        serialized_editable_future_graph = JSONDeserializer().deserialize(
+            JSONSerializer().serialize(editable_future_graph)
+        )
+        serialized_updated_source_graph = JSONDeserializer().deserialize(
+            JSONSerializer().serialize(updated_source_graph)
+        )
+
+        self._compare_serialized_updated_source_graph_and_serialized_editable_future_graph(
+            serialized_updated_source_graph, serialized_editable_future_graph
+        )
+
+        # the source_graph nodegroup and the editable_future_graph nodegroup have been deleted
+        self.assertEqual(nodegroup_count_before, models.NodeGroup.objects.count())
+
+    def test_can_update_graph_slug(self):
+        source_graph = Graph.new(name="TEST RESOURCE", is_resource=True, author="TEST")
+        source_graph.save()
+        editable_future_graph = source_graph.create_editable_future_graph()
+
+        editable_future_graph.slug = "test-resource"
+        editable_future_graph.save()
```

Comment:
I was thinking about the scenarios in which I had this fail and realized I was missing one critical detail - the source_graph should have a slug, but also there should be changes to the editable_future_graph. Here I've copied the node change from `test_update_node` to illustrate.

```suggestion
        source_graph = Graph.new(name="TEST RESOURCE", is_resource=True, author="TEST")
        source_graph.slug = "test-resource"
        source_graph.save()
        editable_future_graph = source_graph.create_editable_future_graph()

        editable_future_graph.append_branch(
            "http://www.cidoc-crm.org/cidoc-crm/P1_is_identified_by",
            graphid=self.NODE_NODETYPE_GRAPHID,
        )
        editable_future_graph.save()
```

Reply:
resolved! Looks like it was missing a str/uuid conversion in the graph validate method.